src/util/triplea/MapXMLCreator/MapXMLCreator.java

```diff
@@ -0,0 +1,839 @@
+		bPreviewHTML.setPreferredSize(new Dimension(300, 30));
+		bPreviewHTML.setAction(new AbstractAction("Preview HTML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				showHTML(MapXMLHelper.s_notes, "HTML Preview");
+			}
+		});
+		m_stepActionPanel.add(bPreviewHTML, gbc_lSecondRow);
+
+		GridBagConstraints gbc_lThirdRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lThirdRow.gridy = 2;
+		m_stepActionPanel.add(new JLabel("<html><big>Congratulation!</big></html>"), gbc_lThirdRow);
+		
+		GridBagConstraints gbc_lFourthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFourthRow.gridy = 3;
+		m_stepActionPanel.add(new JLabel("<html><nobr>You have completed all the steps for creating the map XML.</nobr></html>"), gbc_lFourthRow);
+		
+		final JButton bSave = new JButton("Save Entries to XML");
+		bSave.setPreferredSize(new Dimension(600, 35));
+		bSave.setAction(new AbstractAction("Save Map XML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				MapXMLHelper.saveXML();
+			}
+		});
+		GridBagConstraints gbc_lFifthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFifthRow.insets = new Insets(5, 0, 0, 0);
+		gbc_lFifthRow.gridy = 4;
+		m_stepActionPanel.add(bSave, gbc_lFifthRow);
+	}
+	
+	protected void layoutTabbedPaneWith(final Set<String> keySet)
+	{
+		m_stepActionPanel.setLayout(new BorderLayout());
+		final JTabbedPane tabbedPane = new JTabbedPane(JTabbedPane.SCROLL_TAB_LAYOUT);
+		for (final String key : keySet)
+		{
+			final JPanel innerTabPane = new JPanel();
+			tabbedPane.addTab(key, innerTabPane);
+			ProductionFrontiersPanel.layout(this, innerTabPane, key);
+		}
+		m_stepActionPanel.add(tabbedPane, BorderLayout.CENTER);
+	}
+	
+	
+	/**
+	 * @
+	 * 
+	 * @see HtmlEditorKitTest {@link}https://github.com/KaBoKu/PI/blob/master/WPProjekt/src/kus/swing/HtmlEditorKitTest.java
+	 */
+	public void showHTML(final String htmlString, final String title)
+	{
+		SwingUtilities.invokeLater(new Runnable()
+		{
+			public void run()
+			{
+				// create jeditorpane
+				JEditorPane jEditorPane = new JEditorPane();
+				
+				// make it read-only
+				jEditorPane.setEditable(false);
+				
+				// create a scrollpane; modify its attributes as desired
+				JScrollPane scrollPane = new JScrollPane(jEditorPane);
+				
+				// add an html editor kit
+				HTMLEditorKit kit = new HTMLEditorKit();
+				jEditorPane.setEditorKit(kit);
+				
+				// create a document, set it on the jeditorpane, then add the html
+				Document doc = kit.createDefaultDocument();
+				jEditorPane.setDocument(doc);
+				jEditorPane.setText(htmlString);
+				
+				// now add it all to a frame
+				JFrame j = new JFrame(title);
+				j.getContentPane().add(scrollPane, BorderLayout.CENTER);
+				
+				// make it easy to close the application

```

Comment:
double check how much of the commentary is needed, this comment is really restating what the line of code very clearly says. The comment does not add enough value to justify the maintenance burden of having it around (it's one more thing to check+update if you were to modify this code)